### PR TITLE
feat(Tabs, NavTabs): Add support for overwriting `TextTransform`.

### PR DESCRIPTION
**Pull request by nikitaorliak-cengage**
Issue: #1107

## What I did

Add support for overwriting `TextTransform` for `Tabs` and `NavTabs`.

## Checklist 
- [x] changeset has been added
- [x] Pull request description is descriptive
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works

## How to test

Open Tabs/NavTabs in docs/storybook -> Set `TextTransform` property -> Check styles.

### CODE REVIEW

File: website/react-magma-docs/src/pages/api/nav-tabs.mdx [hunk 1]
```diff
@@ -187,6 +187,65 @@ export function Example() {
...
+          First lowercase
+        </NavTab>
+        <NavTab to="#">Second lowercase</NavTab>
+      </NavTabs>
+    </>
+  );
+}
+```
+
+#### For a certain Tab
```
Comment (silvalaura) [inline website/react-magma-docs/src/pages/api/nav-tabs.mdx:224]: @orion-cengage do we want to support this, or should this style be set the same for all NavTabs?

Comment (orion-cengage) [reply website/react-magma-docs/src/pages/api/nav-tabs.mdx:224]: Does this mean you have the ability to set it on all tabs at the top level, and then "override" that prop on a specific tab if necessary? I could see that being useful for something like:

DATABASE     PERIODICALS     eBOOK

Comment (orion-cengage) [reply website/react-magma-docs/src/pages/api/nav-tabs.mdx:224]: Never mind, I just saw it in action in the docs. I really like that!

image

**Review by orion-cengage — APPROVED** [review]

**Review by silvalaura — APPROVED** [review]
Code looks good and can be merged after the changeset gets updated 👍

**Pull request merged**